pytensor/tensor/pad.py

```diff
@@ -0,0 +1,384 @@
+        )
+
+        # Reverse the direction of the ramp for the "right" side
+        right_ramp = right_ramp[_slice_at_axis(slice(None, None, -1), axis)]  # type: ignore
+
+        padded = set_subtensor(padded[left_slice], left_ramp)
+        padded = set_subtensor(padded[right_slice], right_ramp)
+
+    return padded
+
+
+def flip(x, axis=None):
+    if axis is None:
+        index = ((slice(None, None, -1)),) * x.ndim
+    else:
+        if isinstance(axis, int):
+            axis = [axis]
+        index = [
+            slice(None, None, -1) if i in axis else slice(None, None, None)
+            for i in range(x.ndim)
+        ]
+    return x[index]
+
+
+def _looping_pad(
+    x: TensorVariable, pad_width: TensorVariable, kind: str
+) -> TensorVariable:
+    pad_width = broadcast_to(pad_width, as_tensor((x.ndim, 2)))
+
+    for axis in range(x.ndim):
+        if kind == "wrap":
+
+            def inner_func(i, x):
+                return x
+
+        elif kind == "symmetric":
+            # Delay creation of this function to here because we want to use the axis global inside the scan
+            def inner_func(i, x):
+                return switch(eq(i % 2, 0), flip(x, axis=axis), x)
+
+        size = x.shape[axis]
+        repeats, (left_remainder, right_remainder) = pt_divmod(pad_width[axis], size)
+
+        left_trim = size - left_remainder
+        right_trim = size - right_remainder
+        total_repeats = repeats.sum() + 3  # left, right, center
+
+        parts, _ = scan(inner_func, non_sequences=[x], sequences=arange(total_repeats))
+
+        parts = moveaxis(parts, 0, axis)
+        new_shape = [-1 if i == axis else x.shape[i] for i in range(x.ndim)]
+        x = parts.reshape(new_shape)
+        trim_slice = _slice_at_axis(slice(left_trim, -right_trim), axis)
+        x = x[trim_slice]
+
+    return x
+
+
+def pad(x: TensorLike, pad_width: TensorLike, mode: PadMode = "constant", **kwargs):
+    allowed_kwargs = {
+        "edge": [],
+        "wrap": [],
+        "constant": ["constant_values"],
+        "linear_ramp": ["end_values"],
+        "maximum": ["stat_length"],
+        "mean": ["stat_length"],
+        "median": ["stat_length"],
+        "minimum": ["stat_length"],
+        "reflect": ["reflect_type"],
+        "symmetric": ["reflect_type"],
+    }
+
+    if any(value not in allowed_kwargs[mode] for value in kwargs.keys()):
+        raise ValueError(
+            f"Invalid keyword arguments for mode '{mode}': {kwargs.keys()}"
+        )
+    x = as_tensor(x)
+    pad_width = as_tensor(pad_width)
+
+    if mode == "constant":
+        constant_values = as_tensor(kwargs.pop("constant_values", 0))
+        return _constant_pad(x, pad_width, constant_values)
+    elif mode == "edge":
+        return _edge_pad(x, pad_width)
+    elif mode in ["maximum", "minimum", "mean", "median"]:
+        if mode == "median":
+            # TODO: pt.quantile? pt.median?
```

Comment:
There's an open issue, link to it? https://github.com/pymc-devs/pytensor/issues/53